server/main.go

```diff
@@ -116,10 +116,15 @@ func NameHandler(w http.ResponseWriter, r *http.Request) {
 
 	fstat, err := file.Stat()
 	handlePanic(err)
-	log.Infof("at=send file=%s size=%d", shimmedBuildpack, fstat.Size())
-	w.Header().Add("Content-Type", "application/x-gzip")
-	http.ServeFile(w, r, shimmedBuildpack)
-	log.Infof("at=success file=%s", shimmedBuildpack)
+	if err != nil {
```

Comment:
This will only fix any issue that happens during `Stat()`. What do you think about replacing `handlePanic` with `handleError(err, responseWriter)` instead? This whole panic driven flow is something I've not seen a lot.

Usually it would be something like
```
if err != nil {
  handleErr(err, w)
  return
}
```

Or, if you want to use panics still we need to be prepared to handle all panics
```
defer func() {
        recover()
        w.WriteHeader(http.StatusInternalServerError)
}()
...
fstat, err := file.Stat()
handlePanic(err)
```